Source/WebCore/rendering/BorderPainter.cpp

```diff
@@ -41,6 +41,25 @@
 
 namespace WebCore {
 
+static bool borderStyleFillsBorderArea(BorderStyle style)
+{
+    return !(style == BorderStyle::Dotted || style == BorderStyle::Dashed || style == BorderStyle::Double);
```

Comment:
I think this would be less hazardous (and easier to read) as a switch statement with all cases enumerated.

Anyone who adds a new BorderStyle in the future should be thinking about this (even if the fallback path of painting the mask-image into an image would also work in this case).